BEIMA.Backend/MongoService/MongoConnector.cs

```diff
@@ -0,0 +1,192 @@
+        }
+
+        /*
+         * Checks if the client has been instantiated.
+         * Parameter: none
+         * Returns: true if client is not null, false if client is null.
+         */
+        public bool IsConnected()
+        {
+            return (client != null);
+        }
+
+        /*
+         * Inserts a device into the "devices" collection
+         * Parameter: BsonDocument that contains the fully formed device document (including all required and optional fields)
+         * Returns: ObjectId of the newly inserted object if successful, null if failed
+         */
+        public ObjectId? InsertDevice(BsonDocument doc)
+        {
+            if (!IsConnected())
+            {
+                throw new Exception("MongoConnector is not currently connected");
+            }
+
+            try
+            {
+                var db = client.GetDatabase(dbName);
+                var devices = db.GetCollection<BsonDocument>(deviceCollection);
+                devices.InsertOne(doc);
+                return (ObjectId) doc["_id"];
+            }
+            catch (Exception ex)
+            {
+                Console.Error.WriteLine(ex.Message);
+                return null;
+            }
+        }
+
+        /*
+         * Gets a device from the "devices" collection, given an objectID.
+         * Parameter: objectId, corresponds to the "_id" field for a given document inside of MongoDB
+         * Returns: BsonDocument that was requested
+         */
+        public BsonDocument GetDevice(ObjectId objectId)
+        {
+            if (!IsConnected())
+            {
+                throw new Exception("MongoConnector is not currently connected");
+            }
+
+            var filter = Builders<BsonDocument>.Filter.Eq("_id", objectId);
+
+            try
+            {
+                var db = client.GetDatabase(dbName);
+                var devices = db.GetCollection<BsonDocument>(deviceCollection);
+                return devices.Find(filter).FirstOrDefault();
+            }
+            catch (Exception ex)
+            {
+                Console.Error.WriteLine(ex.Message);
+                return null;
+            }
+        }
+
+        /*
+         * Gets a device from the "devices" collection, given an objectID.
+         * Parameter: objectId, corresponds to the "_id" field for a given document inside of MongoDB
+         * Returns: BsonDocument that was requested
+         */
+        public List<BsonDocument> GetAllDevices()
+        {
+            if (!IsConnected())
+            {
+                throw new Exception("MongoConnector is not currently connected");
+            }
+
+            var filter = Builders<BsonDocument>.Filter.Empty;
+
+            try
+            {
+                var db = client.GetDatabase(dbName);
+                var devices = db.GetCollection<BsonDocument>(deviceCollection);
+                var docs = devices.Find(filter).ToList();
```

Comment:
Is this filter necessary?

Reply:
Yes, an empty filter will return all objects. This is the generally accepted way to do this.